### PR TITLE
7 and 8 implement redux and setup action for creating user rb dh

### LINKED ISSUE #7: Implement Redux
- [x] install all necessary dependencies for redux

### LINKED ISSUE #8: Set Up Redux Action for Creating User
- [x] implement `action creator` for `creating user`
- [x] test action

### PR DESCRIPTION
**Pull request by RajaaBoulassouak**
#### Description
- Install Redux and React Redux dependencies (references #7)
- Implement Create, Sign In and Sign Out User Actions in src/actions/index.js file (references #8)
- Add test for Create, Sign In and Sign Out User Actions in scr/actions/index.test.js
- Set login state to switch based on action in scr/reducers/userReducer.js file (WIP, references #9)

----
#### Issues # and Action
- closes #7 
- closes #8 
----
#### TO DO / Noteworthy
- Login state functionality in reducers will be completed in its own branch 
----
#### Features
- [x] New Feature
- [ ] Expand Feature
- [ ] Optimize/Improve Feature

----
#### Testing
- [ ] No Testing
- [x] Some Testing
- [ ] Sufficient Testing

- [ ] Left notes near functions that have not been tested
- [ ] Intentionally skipped / failing / empty tests as a reminder to add/improve these later

- [ ] Tests have been changed
- [ ] Existing tests are now failing / skipped

----
#### Deployment
- [ ] This is a complete feature, ready for deployment
- [x] This is a partial feature, NOT ready for deployment

----
#### API Changes
(please detail in description)
- [ ] New API
- [ ] Existing API, New Endpoint
- [ ] README.md is updated

----
#### Tech Changes
(please describe)
- [x] New Functional Library/Technology
   - Redux and React Redux dependencies are installed and functional
- [ ] New Testing Library
- [ ] New Debugging Tool
- [ ] README.md is updated

### CODE REVIEW

**Comment by RajaaBoulassouak** [comment]
Hi Derek. I updated the PR template name in master. It wasn't rendering because I had not written out 'pull request'. It's working now how you can see ;-). 
I'm not sure if card 8 is complete, but if not or if we need add more user info later, I thought we can reopen it in the project board. 
And Yay, looks like Circle CI is doing it's job. Nice work!

File: src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    password
+  }
+}
+
+export const signOutUser = () => {
+  return {
+    type: 'SIGN_OUT_USER'
+  }
+}
+
```
Comment (drhanciles) [inline src/actions/index.js:24]: Greate work on these actions Rajaa!

File: src/reducers/userReducer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    case 'SIGN_OUT_USER': 
+      return {
+        state.id: ''
+      }
+    default: 
+      return state
+  }
+}
+
+export default user
```
Comment (drhanciles) [inline src/reducers/userReducer.js:21]: Great work on this Rajaa! After reviewing over how we would want to the store our user object, lets also add an id, into the payload for when we sign a user in.

File: src/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    password
+  }
+}
+
+export const signInUser = (email, password) => {
+  return {
+    type: 'SIGN_IN_USER', 
+    email, 
+    password
+  }
```
Comment (drhanciles) [inline src/actions/index.js:16]: This is great, but we should also add an id in the payload for when we dispatch the `signInUser` action.

**Comment by drhanciles** [comment]
Great work on this Rajaa I added some comments above once we get them changed, i'll go ahead and merge this in!

**Comment by RajaaBoulassouak** [comment]
Thank you for the feedback, Derek! The id attribute is now added on both suggested spots. Let me know if there is anything else that we can add/change before merging.

**Comment by drhanciles** [comment]
Sweet this looks good to be merged in!

**Pull request merged**